pkg/client/userd/trafficmgr/agents.go

```diff
@@ -51,6 +58,30 @@ func (s *session) getCurrentAgentsInNamespace(ns string) map[string]*manager.Age
 	return agents
 }
 
+func (s *session) getCurrentSidecarsInNamespace(ctx context.Context, ns string) map[string]*agentconfig.Sidecar {
+	sidecars := make(map[string]*agentconfig.Sidecar)
```

Comment:
Place this line after successful retrieval of `cm`. Then preallocate so that the map doesn't need to grow.
```suggestion
	sidecars := make(map[string]*agentconfig.Sidecar, len(cm.Data))
```